### PR TITLE
Feature/health monitor camera driver

### PR DESCRIPTION
**Pull request by lewisid**
# PR Details
## Description

Additional logic has been added to include the camera as a required driver in the health monitor node. The driver_manager has been restructured and test cases have been added to confirm that the system recognizes the camera's driver status alongside the Lidar and GPS drivers and sends the appropriate alert messages to the health monitor.
## Related Issue

https://github.com/usdot-fhwa-stol/carma-platform/issues/1072#
## Motivation and Context

The health_monitor node currently only requires Lidar, GPS, and Controller drivers. It is time to expand this to other drivers needed by the system, specifically the camera driver used for object perception.
## How Has This Been Tested?
This has been unit tested locally on a VM using gtest/gmock.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [x] I have added any new packages to the sonar-scanner.properties file
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by msmcconnell** [comment]
@lewisid wait on resolving these merge conflicts until vanden-plas is merged back to develop. I think many of them will go away for this PR then.

File: health_monitor/test/test_plugin_manager.cpp
Comment (msmcconnell) [inline health_monitor/test/test_plugin_manager.cpp:55]: Why did this test case change from false to true? Your changes should not be side affecting plugin_manager behavior.

**Review by msmcconnell — CHANGES_REQUESTED** [review]

File: health_monitor/test/test_plugin_manager.cpp
Comment (lewisid) [reply health_monitor/test/test_plugin_manager.cpp:55]: I don't really know. When I ran the test cases in VSCode, I noticed that one failed for some reason.

**Review by msmcconnell — APPROVED** [review]

**Pull request merged**
